LEAF_Request_Portal/js/formSearch.js

```diff
@@ -935,57 +940,44 @@ var LeafFormSearch = function (containerID) {
-                    dataType: "json",
-                    success: function (res) {
-                        let allStepsData = res;
-                        var categories =
-                            '<select id="' +
-                            prefixID +
-                            "widgetMat_" +
-                            widgetID +
-                            '" class="chosen" aria-label="stepID" style="width: 250px">';
-                        categories +=
-                            '<option value="submitted">Submitted</option>';
-                        categories +=
-                            '<option value="deleted">Cancelled</option>';
-                        categories +=
-                            '<option value="resolved" selected>Resolved</option>';
-                        categories +=
-                            '<option value="actionable">Actionable by me</option>';
-                        //categories += '<option value="destruction">Scheduled for Destruction</option>';
-                        for (var i in allStepsData) {
-                            categories +=
-                                '<option value="' +
-                                allStepsData[i].stepID +
-                                '">' +
-                                allStepsData[i].description +
-                                ": " +
-                                allStepsData[i].stepTitle +
-                                "</option>";
-                        }
-                        categories += "</select>";
-                        // quick and dirty fix to avoid a race condition related to common
-                        // implementations of formSearch. Since the new default UI will trigger
-                        // the parent ajax call, we don't want to overwrite the existing widget.
-                        // This can cause a minor UX issue where changing the search term from anything
-                        // else to a "Current Status" would result in a missing dropdown of status options.
-                        if($("#" + prefixID + "widgetMatch_" + widgetID).html() == "") {
-                            $("#" + prefixID + "widgetMatch_" + widgetID).html(
-                                categories
-                            );
+                        ? "./api/workflow/steps?x-filterData=workflowID,stepID,stepTitle,description"
+                        : rootURL + "api/workflow/steps?x-filterData=workflowID,stepID,stepTitle,description";
+                if(cache['api/workflow/steps'] == undefined) {
+                    cache['api/workflow/steps'] = $.ajax({
+                        type: "GET",
+                        url,
+                        dataType: "json"
+                    }).then(res => {
+                        // Hide standard workflows unless the GET parameter "dev" exists
+                        if(new URLSearchParams(window.location.search).get('dev') == null) {
```

Comment:
I'm thinking this needs a fail/catch.  If it gets something weird back from the network it could have odd consequences down the line.